src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -338,66 +333,120 @@ func KafkaAuthConfig(
 				sarama.SASLTypeSCRAMSHA512,
 			)
 		}
-		return &config.SCRAM{
-			User:     u,
-			Password: p,
-			Type:     m,
+		return &config.SASL{
+			User:      u,
+			Password:  p,
+			Mechanism: m,
 		}, nil
 	}
 }
 
-func BuildTLSConfig(
-	certFile,
-	keyFile,
-	truststoreFile *string,
+func BuildAdminApiTLSConfig(
+	certFile, keyFile, truststoreFile *string,
+	configuration func() (*config.Config, error),
 ) func() (*config.TLS, error) {
 	return func() (*config.TLS, error) {
-		c := *certFile
-		k := *keyFile
-		t := *truststoreFile
+		defaultVal := func() (*config.TLS, error) {
+			conf, err := configuration()
+			if err != nil {
+				return nil, err
+			}
+			return conf.Rpk.AdminApi.TLS, nil
+		}
+		return buildTLS(
+			certFile,
+			keyFile,
+			truststoreFile,
+			"REDPANDA_ADMIN_TLS_CERT",
+			"REDPANDA_ADMIN_TLS_KEY",
+			"REDPANDA_ADMIN_TLS_TRUSTSTORE",
+			defaultVal,
+		)
+	}
+}
 
-		if c == "" {
-			c = os.Getenv("REDPANDA_TLS_CERT")
-		}
-		if k == "" {
-			k = os.Getenv("REDPANDA_TLS_KEY")
-		}
-		if t == "" {
-			t = os.Getenv("REDPANDA_TLS_TRUSTSTORE")
-		}
+func BuildKafkaTLSConfig(
+	certFile, keyFile, truststoreFile *string,
+	configuration func() (*config.Config, error),
+) func() (*config.TLS, error) {
+	return func() (*config.TLS, error) {
+		defaultVal := func() (*config.TLS, error) {
+			conf, err := configuration()
+			if err != nil {
+				return nil, err
+			}
+			return conf.Rpk.KafkaApi.TLS, nil
```

Comment:
I don't think this is enough: the `BuildTLS` function in pkg/tls/tls.go does not initialize a `*tls.Config` if the cert/key/ca are all empty. That function needs to be changed so that it allows creating a client with a default tls.Config.

Reply:
This is changed in a later commit